src/test/java/io/lettuce/core/cluster/pubsub/RedisClusterPubSubConnectionIntegrationTests.java

```diff
@@ -101,6 +112,32 @@ void testRegularClientPubSubShardChannels() {
         assertThat(channelsOnOtherNode).isEmpty();
     }
 
+    @Test
+    void subscribeToShardChannel() throws Exception {
+        pubSubConnection.addListener(this);
+        pubSubConnection.async().ssubscribe(shardChannel);
+        assertThat(shardChannels.poll(3, TimeUnit.SECONDS)).isEqualTo(shardChannel);
+    }
+
+    @Test
+    void subscribeToShardChannelViaOtherEndpoint() throws Exception {
```

Comment:
The test should be guarded with `@EnabledOnCommand("SSUBSCRIBE)`

Reply:
done